helper.php

```diff
@@ -364,6 +386,156 @@ function _formatData($column, $value, Doku_Renderer_xhtml $R) {
     }
 
     /**
+     * Return formated data, depending on column type
+     *
+     * @param array               $column
+     * @param string              $value
+     * @param Doku_Renderer       $R
+     * @return string
+     */
+    function _formatDataNew($column, $value, Doku_Renderer $R) {
+        global $conf;
+        $vals = explode("\n", $value);
+        $outs = array();
+
+        //multivalued line from db result for pageid and wiki has only in first value the ID
+        $storedID = '';
+
+        foreach($vals as $val) {
+            $val = trim($val);
+            if($val == '') continue;
+
+            $type = $column['type'];
+            if(is_array($type)) {
+                $type = $type['type'];
+            }
+            switch($type) {
+                case 'page':
+                    $val = $this->_addPrePostFixes($column['type'], $val);
+                    $val = $this->ensureAbsoluteId($val);
+                    $outs[] = $R->internallink($val, null, null, true);
+                    break;
+                case 'title':
+                    list($id, $title) = explode('|', $val, 2);
+                    $id = $this->_addPrePostFixes($column['type'], $id);
+                    $id = $this->ensureAbsoluteId($id);
+                    $outs[] = $R->internallink($id, $title, null, true);
+                    break;
+                case 'pageid':
+                    list($id, $title) = explode('|', $val, 2);
+
+                    //use ID from first value of the multivalued line
+                    if($title == null) {
+                        $title = $id;
+                        if(!empty($storedID)) {
+                            $id = $storedID;
+                        }
+                    } else {
+                        $storedID = $id;
+                    }
+
+                    $id = $this->_addPrePostFixes($column['type'], $id);
+
+                    $outs[] = $R->internallink($id, $title, null, true);
+                    break;
+                case 'nspage':
+                    // no prefix/postfix here
+                    $val = ':' . $column['key'] . ":$val";
+
+                    $outs[] = $R->internallink($val, null, null, true);
+                    break;
+                case 'mail':
+                    list($id, $title) = explode(' ', $val, 2);
+                    $id = $this->_addPrePostFixes($column['type'], $id);
+                    if(!$title) {
+                        $title = $id;
+                    } else {
+                        $title = hsc($title);
+                    }
+                    $outs[] = $R->emaillink($id, $title, true);
```

Comment:
The  _formatDataOld() has some rawurlencoding applied, which is applied to the content of the `title` attribute as well. For the title attribute this encoding is not required.
My proposal is to improve the unittest and to modify the behavior in the _formatDataOld() such that it matches the new behavior.
```php
                     if($conf['mailguard'] == 'visible') {
                         $id = rawurlencode($id);
                     }
                     $outs[] = '<a href="mailto:' . $id . '" class="mail" title="' . $id . '">' . $title . '</a>';
```